libs/features/personalization/personalization.js

```diff
@@ -1157,8 +1167,23 @@ export async function init(enablements = {}) {
     if (pzn) loadLink(getXLGListURL(config), { as: 'fetch', crossorigin: 'anonymous', rel: 'preload' });
   }
 
-  if (target === true) manifests = manifests.concat(await callMartech(config));
-  if (target === 'postlcp') callMartech(config);
+  if (target === true) {
+    let localManifests = [];
+    if (enablePersonalizationV2()) {
+      localManifests = await callMartech(config, targetInteractionPromise);
+    } else {
+      localManifests = await callMartech(config);
+    }
+    manifests = manifests.concat(localManifests);
+  }
+
+  if (target === 'postlcp') {
+    if (enablePersonalizationV2()) {
+      await callMartech(config, targetInteractionPromise);
+    } else {
+      callMartech(config);
+    }
+  }
```

Comment:
This can be combined, untested demo code:
```suggestion
if (target === true || target === 'postlcp') {
  const callMartechPromise = enablePersonalizationV2() 
    ? callMartech(config, targetInteractionPromise) 
    : callMartech(config);

  if (target === true) {
    let localManifests = await callMartechPromise;
    manifests = manifests.concat(localManifests);
  } else if (target === 'postlcp') {
    await callMartechPromise;
  }
}
```